main/global-accelerators.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+const {globalShortcut} = require('electron');
+const ipc = require('electron-better-ipc');
+const shortcutToAccelerator = require('../renderer/utils/shortcut-to-accelerator');
+const store = require('./common/settings');
+const {openCropperWindow} = require('./cropper');
+
+const registerShortcut = shortcut => {
+  try {
+    const shortcutAccelerator = shortcutToAccelerator(shortcut);
+    globalShortcut.register(shortcutAccelerator, openCropperWindow);
+  } catch (error) {
+    console.error('Error registering shortcut', shortcut);
+  }
+};
+
+const registrerFromStore = () => {
+  if (store.get('recordKeyboardShortcut')) {
+    const cropperShortcut = store.get('cropperShortcut');
+    if (cropperShortcut) {
+      registerShortcut(cropperShortcut);
+    }
+  } else {
+    globalShortcut.unregisterAll();
+  }
+};
+const initializeGlobalAccelerators = () => {
+  ipc.answerRenderer('update-shortcut', ({setting, shortcut}) => {
+    try {
+      if (store.has(setting)) {
+        const oldShortcut = store.get(setting);
+        const shortcutAccelerator = shortcutToAccelerator(oldShortcut);
+        if (globalShortcut.isRegistered(shortcutAccelerator)) {
+          globalShortcut.unregister(shortcutAccelerator);
+        }
+      }
+    } catch (error) {
+      console.error('Error unregestering old shortcutAccelerator', error);
+    }
+    if (shortcut) {
+      registerShortcut(shortcut);
+    }
+    store.set(setting, shortcut);
+  });
+
+  // Register keyboard shortcuts from store
+  registrerFromStore();
+
+  store.events.on('change', registrerFromStore);
```

Comment:
Can use [onKeyChange](https://github.com/sindresorhus/electron-store#ondidchangekey-callback) to only register again if the specific key changes, and not if any store key changes.

Also if we listen for changes like this, we probably don't even need the `ipc` event as long as we update the setting in the store in the prefs window

Reply:
Yeah, I was thinking that too but I didn't know about onKeyChange! Nice!